File: src/main/java/com/warrantybee/api/enumerations/NotificationType.java
package com.warrantybee.api.enumerations;

import com.warrantybee.api.enumerations.interfaces.IEnumeration;
import lombok.Getter;

/**
 * Defines the different types of notifications that can be sent to users,
 * such as OTP messages, welcome emails, or password change alerts.
 */
@Getter
public enum NotificationType implements IEnumeration {

    /** Default or unspecified notification type. */
    NONE(0),

    /** Notification sent when an OTP is requested for multi-factor authentication (login). */
    MFA_LOGIN(1),

    /** Notification sent when an OTP is requested for password recovery. */
    FORGOT_PASSWORD(2),

    /** Notification sent to welcome a new user upon registration. */
    WELCOME(3),

    /** Notification sent when a user successfully changes their password. */
    PASSWORD_CHANGED(4);

    /** The unique integer code associated with this notification type. */
    private final int code;

    /**
     * Constructs a {@code NotificationType} with the specified integer code.
     *
     * @param code the unique code representing this notification type
     */
    NotificationType(int code) {
        this.code = code;
    }
}

File: src/WarrantyBee.Domain/Enums/NotificationType.cs
namespace WarrantyBee.Domain.Enums;

/// <summary>
/// Represents the types of notifications that can be sent to a user.
/// </summary>
public enum NotificationType
{
    /// <summary>
    /// No notification type specified.
    /// </summary>
    None = 0,
    /// <summary>
    /// Multi-factor authentication login notification.
    /// </summary>
    MfaLogin = 1,
    /// <summary>
    /// Forgot password notification.
    /// </summary>
    ForgotPassword = 2,
    /// <summary>
    /// Welcome notification for new users.
    /// </summary>
    Welcome = 3,
    /// <summary>
    /// Notification indicating a password change.
    /// </summary>
    PasswordChanged = 4
}
